src/components/sectionListRe.rei

```diff
@@ -1,68 +1,76 @@
-type section 'item =
-  Js.t {
-    .
-    data : array 'item, key : Js.Undefined.t string, renderItem : Js.Undefined.t (renderItem 'item)
-  }
-and renderItem 'item =
-  Js.t {
-    .
-    item : 'item,
-    index : int,
-    section : section 'item,
-    separators :
+module Make:
+  (Item: {type item;}) =>
+  {
+    type section =
```

Comment:
We should decide if we want to expose JS primitives in the public API. It's mixed now. In some places they are converted to records, in others they remain JS. I'm personally in favour of converting them to records, but also, what's more important, keeping it consistent.